dom/test/browser/src/elements.ts

```diff
@@ -0,0 +1,146 @@
+import * as simulant from 'simulant';
+import * as assert from 'assert';
+import isolate from '@cycle/isolate';
+import xs, {Stream, MemoryStream} from 'xstream';
+import delay from 'xstream/extra/delay';
+import concat from 'xstream/extra/concat';
+import {setup, run} from '@cycle/run';
+import {
+  svg,
+  div,
+  span,
+  h2,
+  h3,
+  h4,
+  p,
+  makeDOMDriver,
+  MainDOMSource,
+  VNode,
+} from '../../../lib';
+
+function createRenderTarget(id: string | null = null) {
+  const element = document.createElement('div');
+  element.className = 'cycletest';
+  if (id) {
+    element.id = id;
+  }
+  document.body.appendChild(element);
+  return element;
+}
+
+describe('DOMSource.elements()', function() {
+  it('should return a stream of documents when querying "document"', function(
+    done,
+  ) {
+    function app(sources: {DOM: MainDOMSource}) {
+      return {
+        DOM: xs.of(div('.top-most', [p('Foo'), span('Bar')])),
+      };
+    }
+
+    const {sinks, sources, run} = setup(app, {
+      DOM: makeDOMDriver(createRenderTarget()),
+    });
+
+    function isDocument(element: any): element is Document {
+      return 'body' in element && 'head' in element;
+    }
+
+    let dispose: any;
+    sources.DOM.select('document').elements().take(1).addListener({
+      next: root => {
```

Comment:
Do these tests pass? I think `root` should be an array here since `elements()` was used